lib/google_search_data_viewer/keywords/keyword.ex

```diff
@@ -0,0 +1,44 @@
+defmodule GoogleSearchDataViewer.Keywords.Keyword do
+  import Ecto.Query, warn: false
+
+  alias GoogleSearchDataViewer.Keywords.Schemas.KeywordUpload
+  alias GoogleSearchDataViewer.Repo
+
+  def insert_keyword_uploads(attrs) do
+    Repo.insert_all(KeywordUpload, attrs)
+  end
+
+  def create_keyword_uploads(keywords, user) do
+    current_date_time = NaiveDateTime.truncate(NaiveDateTime.utc_now(), :second)
+
+    keywords
+    |> Enum.map(fn keyword ->
+      create_params_for_keyword_and_user(keyword, user.id)
+    end)
+    |> Enum.map(fn params -> create_changeset_and_parse(params) end)
+    |> Enum.map(&Map.from_struct/1)
+    |> Enum.map(fn params -> Map.drop(params, [:__meta__, :user, :id]) end)
+    |> Enum.map(fn params -> insert_timestamps(params, current_date_time) end)
```

Comment:
This can be process in one run so 
```suggestion
    |> Enum.map(fn keyword -> 
      # These operations can wrap in private function
      create_params_for_keyword_and_user(keyword, user.id)
      |> create_changeset_and_parse(params) 
      |> Map.from_struct()
      |> Map.drop([:__meta__, :user, :id]) 
      |> insert_timestamps(current_date_time)
    end)
```
And as mentioned in the comment of the above code example, we can further refactor a bit to add a level of abstraction to this function by moving to a private function. IMO would make the code more readable, let me know your thought 💭 
```suggestion
    keywords
    |> Enum.map(&process_keyword_params/1)
    |> insert_keyword_uploads()
    |> elem(0)
```

Reply:
Yes I think it's a little easier to read the flow with applying all the changes with the initial Map function.

Yup, that's a good idea. Separating it into its own function `process_keyword_params` is better than having the parsing/restructering in the function definition `create_keyword_uploads` This more accurately reflects what is really happening and separates the two.